src/coreclr/vm/siginfo.cpp

```diff
@@ -3908,7 +3952,59 @@ MetaSig::CompareElementType(
                         pOtherModule,
                         tkOther,
                         ClassLoader::ReturnNullIfNotFound,
-                        ClassLoader::FailIfUninstDefOrRef);
+                        ClassLoader::PermitUninstDefOrRef);
+
+                    // If the return type is a generic type definition, we need to
```

Comment:
I do not understand why this is needed. Is this a left over from some earlier version?

Reply:
This is for the case where the type is fully instantiated in the defintion. 

```csharp
private class A
{
    private List<B> M() => default;
}
```

Reply:
I would expect this case to be covered by the next `ELEMENT_TYPE_GENERICINST` below.

One of the signature is going to have `ELEMENT_TYPE_GENERICINST` and the other signature is going to have `ELEMENT_TYPE_INTERNAL`. We will load the type for `ELEMENT_TYPE_GENERICINST`, and then compare the types for exact match.